ampe.c

```diff
@@ -324,18 +324,23 @@ static int protect_frame(struct candidate *cand, struct ieee80211_mgmt_frame *mg
     unsigned short cat_to_mic_len;
     struct mesh_node *mesh = cand->conf->mesh;
     size_t ampe_ie_len;
+    u8 ftype = mgmt->action.action_code;
     le16 igtk_keyid;
 
     assert(mic_start && cand && mgmt && len);
 
 #define MIC_IE_BODY_SIZE     AES_BLOCK_SIZE
 
-    /* MGTK + RSC + Exp */
-    ampe_ie_len = sizeof(struct ampe_ie) + 16 + 8 + 4;
+    ampe_ie_len = sizeof(struct ampe_ie);
 
-    if (mesh->conf->protected_management_frames) {
-        /* IGTK KeyId + IPN + IGTK */
-        ampe_ie_len += 2 + 6 + 16;
+    if (ftype != PLINK_CLOSE) {
+        /* MGTK + RSC + Exp */
```

Comment:
AMPE IE is available in the mesh peering close frame defined in section 8.5.16.4.2 IEEE802.11-2012.